src/odb/src/upf/upf.cpp

```diff
@@ -217,4 +221,631 @@ bool set_isolation(utl::Logger* logger,
+
+    // Find enable & data pins for the isolation cell
+    odb::dbMTerm* enable_term = nullptr;
+    odb::dbMTerm* data_term = nullptr;
+    odb::dbMTerm* output_term = nullptr;
+    sta::LibertyPort* out_lib_port = nullptr;
+    sta::LibertyPort* enable_lib_port = nullptr;
+    sta::LibertyPort* data_lib_port = nullptr;
+    for (auto&& term : cell_terms) {
+      sta::LibertyPort* lib_port
+          = smallest_iso_l->findLibertyPort(term->getName().c_str());
+
+      if (!lib_port) {
+        continue;
+      }
+
+      if (lib_port->isolationCellData()) {
+        data_term = term;
+        data_lib_port = lib_port;
+      }
+
+      if (lib_port->isolationCellEnable()) {
+        enable_term = term;
+        enable_lib_port = lib_port;
+      }
+
+      if (term->getIoType() == odb::dbIoType::OUTPUT) {
+        output_term = term;
+        out_lib_port = lib_port;
+      }
+    }
+
+    if (!output_term || !data_term || !enable_term) {
+      logger->warn(utl::ODB,
+                   130152,
+                   "Isolation %s cells defined, but can't find one of output, "
+                   "data or enable terms.",
+                   iso->getName());
+      continue;
+    }
+
+    // Determine if an inverter is needed for the control pin & output
+    auto func = out_lib_port->function();
+    int inv_needed = check_isolation_match(
+        func, enable_lib_port, isolation_sense, isolation_clamp_val);
+
+    // find the smallest possible inverter
+    odb::dbMaster* inverter_m = nullptr;
+    odb::dbMTerm *input_m = nullptr, *output_m = nullptr;
+
+    if (inv_needed > 0
+        && find_smallest_inverter(logger, block, inverter_m, input_m, output_m)
+               == false) {
+      logger->warn(utl::ODB, 130482, "can't find any inverters");
+      continue;
+    }
+
+    // Iterate through all pd's instances and determine if any signal needs
+    // isolation If one is found, create dbInst from the isolation cell +
+    // connect control signal of isolation to dbInst
+    // connect previous net to data input
+    // connect output from dbInst to
+
+    odb::dbGroup* curr_group = domain->getGroup();
+    auto insts = curr_group->getInsts();
+
+    for (auto&& inst : insts) {
+      auto iterms = inst->getITerms();
+      for (auto&& iterm : iterms) {
+        if (iterm->getIoType() != odb::dbIoType::OUTPUT)
+          continue;
+
+        auto net = iterm->getNet();
+
+        if (!net) {
+          continue;
+        }
+
+        auto connectedIterms = net->getITerms();
+
+        if (connectedIterms.size() < 2)
+          continue;
+
+        // Find ITERMS that belong to instances outside of this power domain
+        std::vector<odb::dbITerm*> external_iterms;
+        for (auto&& connectedIterm : connectedIterms) {
+          auto connectedInst = connectedIterm->getInst();
+
+          if (!connectedInst->getGroup()
+              || connectedInst->getGroup()->getName()
+                     != curr_group->getName()) {
```

Comment:
Can you compare the groups rather than their names?  Would two different groups ever have the same name?

Reply:
Not really. I was working with names to avoid working with pointers